### PR TITLE
Sharpness

### PR DESCRIPTION
**Pull request by MarkShleifer**

### CODE REVIEW

File: src/app/packages/menu-items/tools/reducers/tools.reducer.ts [hunk 1]
```diff
@@ -50,6 +50,24 @@ export function ToolsReducer(state = toolsInitialState,action: ToolsActions): IT
...
+		case ToolsActionsTypes.TOGGLE_AUTO_IMAGE_PROCESSING_SUCCESS:
+
+			tmpMap = new Map(state.flags);			
+			tmpMap.set('auto_image_processing',action.payload);
+			return { ...state,flags: tmpMap };
+
+		case ToolsActionsTypes.ENABLE_AUTO_IMAGE_PROCESSING:
+
+			tmpMap = new Map(state.flags);
+			tmpMap.set('auto_image_processing_disabled', false);
```
Comment (yairtawil) [inline src/app/packages/menu-items/tools/reducers/tools.reducer.ts]: think about adding `(auto_image_processing, false)`

File: src/app/app-reducers/effects/tools.app.effect.spec.ts [hunk 2]
```diff
@@ -87,15 +122,168 @@ describe('ToolsAppEffects', () => {
...
+		it('onActiveMapChanges with overlay and image processing as false on should raise EnableImageProcessing and ToggleImageProcessingSuccess with false', () => {
+			const mapId = icaseState.selected_case.state.maps.active_map_id;
+			const active_map = icaseState.selected_case.state.maps.data.find((map) => map.id === mapId);
+			active_map.data.isAutoImageProcessingActive = false;
+
+			effectsRunner.queue(new ActiveMapChangedAction());
+
+			toolsAppEffects.onActiveMapChanges$.subscribe(_result => {
+				const result = _result instanceof EnableImageProcessing || _result instanceof ToggleAutoImageProcessingSuccess;
+				expect(result).toBe(true);
```
Comment (MarkShleifer) [inline src/app/app-reducers/effects/tools.app.effect.spec.ts]: done

File: src/app/app-reducers/effects/tools.app.effect.spec.ts [hunk 3]
```diff
@@ -87,15 +122,168 @@ describe('ToolsAppEffects', () => {
...
+		});
+	});
+
+	describe('onDisplayOverlaySuccess', () => {
+		it('onDisplayOverlaySuccess with image processing as true should raise ToggleMapAutoImageProcessing and ToggleAutoImageProcessingSuccess accordingly', () => {
+			effectsRunner.queue(new DisplayOverlaySuccessAction({ id: 'id' }));
+
+			toolsAppEffects.onActiveMapChanges$.subscribe(_result => {
+				const result = _result instanceof ToggleMapAutoImageProcessing || _result instanceof ToggleAutoImageProcessingSuccess;
+				expect(result).toBe(true);
```
Comment (MarkShleifer) [inline src/app/app-reducers/effects/tools.app.effect.spec.ts]: done

File: src/app/app-reducers/effects/tools.app.effect.spec.ts [hunk 4]
```diff
@@ -87,15 +122,168 @@ describe('ToolsAppEffects', () => {
...
+
+			toolsAppEffects.onSelectCaseById$.subscribe(_result => {
+				expect(_result instanceof DisableImageProcessing).toBeTruthy();
+			});
+		});
+	});
+
+	describe('toggleAutoImageProcessing', () => {
+		it('toggleAutoImageProcessing with image processing as true should raise ToggleMapAutoImageProcessing, UpdateCaseAction and ToggleAutoImageProcessingSuccess accordingly', () => {
+			effectsRunner.queue(new SelectCaseByIdAction('asdfasd'));
```
Comment (MarkShleifer) [inline src/app/app-reducers/effects/tools.app.effect.spec.ts]: done

File: src/app/app-reducers/effects/tools.app.effects.ts [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import { Injectable } from '@angular/core';
```
Comment (MarkShleifer) [inline src/app/app-reducers/effects/tools.app.effects.ts]: done

File: src/app/packages/map-facade/effects/map.effects.ts [hunk 6]
```diff
@@ -48,5 +49,13 @@ export class MapEffects{
 		.ofType(MapActionTypes.COMPOSITE_MAP_SHADOW_ACTION)
 		.share();
 
+	@Effect({dispatch:false})
+	onToggleImageProcessing$: Observable<any> = this.actions$
+		.ofType(MapActionTypes.TOGGLE_MAP_AUTO_IMAGE_PROCESSING)
+		.do((action: ToggleMapAutoImageProcessing) => {
```
Comment (MarkShleifer) [inline src/app/packages/map-facade/effects/map.effects.ts]: done

File: src/app/packages/menu-items/tools/reducers/tools.reducer.ts (hunk 1 above)
Comment (MarkShleifer) [inline src/app/packages/menu-items/tools/reducers/tools.reducer.ts]: done

File: src/app/app-reducers/effects/tools.app.effect.ts [hunk 7]
```diff
@@ -3,53 +3,133 @@ import { Effect, Actions } from '@ngrx/effects';
...
+					new EnableImageProcessing(),
+					new ToggleAutoImageProcessingSuccess(active_map.data.isAutoImageProcessingActive)
+				];
+			}
+		});
+
+	@Effect()
+	onDisplayOverlay$: Observable<any> = this.actions$
+		.ofType(OverlaysActionTypes.DISPLAY_OVERLAY)
+		.map(() => new EnableImageProcessing());
```
Comment (MarkShleifer) [inline src/app/app-reducers/effects/tools.app.effect.ts]: done

**Pull request merged**